x/crosschain/types/cctx.go

```diff
@@ -203,6 +205,51 @@ func (m CrossChainTx) SetOutboundBallotIndex(index string) {
 }
 
 // GetCctxIndexFromBytes returns the CCTX index from a byte array.
+func (m CrossChainTx) GetValidReceiverAddress() (ethcommon.Address, error) {
```

Comment:
Will be adding more unit tests , and wanted to open up this pr for discussions wether this sort of validation based on coin type is the direction we want to go with